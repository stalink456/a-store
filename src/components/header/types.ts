export type HeaderContextType = {
  open?: boolean;
  handleOpenMenu?: () => void;
};
