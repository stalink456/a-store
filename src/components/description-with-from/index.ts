export { DescriptionWithForm } from './description-with-form';
