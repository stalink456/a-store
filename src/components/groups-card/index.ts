export { GroupsCard } from './groups-card';
