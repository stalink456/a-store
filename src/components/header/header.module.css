@import "../../css/variables.css";

.header {
  height: 80px;
  background-color: var(--white-color);
  z-index: 2;
}

.header__container {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: fixed;
  padding: var(--gap-s) var(--gap-m);
  background-color: var(--white-color);
}

.header__container a {
  color: var(--logo-color);
  text-decoration: none;
}

.header__menu button {
  font-size: 30px;
  font-weight: 700;
  color: var(--black-color);
}

@media (max-width: 960px) {
  .header__menu_button {
    display: none;
  }
}
