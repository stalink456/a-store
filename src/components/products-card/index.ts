export { ProductsCard } from "./products-card";
