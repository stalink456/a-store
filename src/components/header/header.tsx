import React, { useRef } from "react";
import { Space } from "@alfalab/core-components/space";
import { Button } from "@alfalab/core-components/button";
import { IconButton } from "@alfalab/core-components/icon-button";
import { GiHamburgerMenu } from "react-icons/gi";
import { Menu } from "components/menu";
import { TitleLink } from "components/ui-components/title-link";
import { useSidePanel } from "hooks/use-side-panel";

import styles from "./header.module.css";

export const Header: React.FC = () => {
  const menuRef = useRef<HTMLDivElement>(null);
  const { open, handleOpenMenu } = useSidePanel(menuRef);

  return (
    <div className={styles.header}>
      <Space direction="horizontal" className={styles.header__container}>
        <Space direction="horizontal">
          <TitleLink
            view="large"
            tag="h1"
            weight="bold"
            text="A-Store"
            route="/"
          />
        </Space>

        <Space className={styles.header__menu} direction="horizontal" size={8}>
          <IconButton
            icon={GiHamburgerMenu}
            view="primary"
            size="xs"
            onClick={handleOpenMenu}
          />
          <Button
            className={styles.header__menu_button}
            view="ghost"
            onClick={handleOpenMenu}
          >
            меню
          </Button>
        </Space>
      </Space>
      <Menu ref={menuRef} open={open} handleOpenMenu={handleOpenMenu} />
    </div>
  );
};
