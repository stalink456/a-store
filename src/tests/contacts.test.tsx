import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Contacts from 'pages/contacts';
import { Header } from 'components/header';

describe('Contacts component', () => {
  it('should render Contacts component', () => {
    const { baseElement } = render(<Contacts />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should render Contacts component with title', async () => {
    render(<Contacts />);

    const result = await screen.findByText('Контакты');

    expect(result).toBeInTheDocument();
  });

  it('should render Header component with transition to Contacts page', async () => {
    render(
      <MemoryRouter>
        <Routes>
          <Route path='/' element={<Header />} />
          <Route path='/contacts' element={<Contacts />} />
        </Routes>
      </MemoryRouter>
    );

    fireEvent.click(await screen.findByText('меню'));
    fireEvent.click(await screen.findByText('Контакты'));

    const result = await screen.findByText('info@alfabankstore.ru');

    expect(result).toBeInTheDocument();
  });

  it('should render Contacts component with non-existent description', async () => {
    render(<Contacts />);

    const result = screen.queryByText('non-existent text') as HTMLImageElement;
    expect(result).toBeFalsy();
  });
});
