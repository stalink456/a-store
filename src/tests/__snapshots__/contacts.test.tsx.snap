// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts component should render Contacts component 1`] = `
<body>
  <div>
    <div
      class="contacts"
    >
      <div
        class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as"
      >
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
          style="margin-bottom: 16px;"
        >
          <div
            class="space__spaceContainer_nr3as space__start_nr3as"
          >
            <div
              class="space__spaceItem_nr3as"
            >
              <h1
                class="typography__component_19zgr contacts__title typography__styrene-xlarge_pxbut typography__bold_pxbut"
              >
                Контакты
              </h1>
            </div>
          </div>
        </div>
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
          style="margin-bottom: 16px;"
        >
          <div
            class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as contacts__info"
          >
            <div
              class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
              style="margin-bottom: 16px;"
            >
              <div
                class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
              >
                <div
                  class="space__spaceItem_nr3as"
                  style="margin-bottom: 0px;"
                >
                  <span
                    class="typography__primary-medium_c16y5 typography__bold_c16y5"
                  >
                    +7 906 061 60 20
                  </span>
                </div>
                <div
                  class="space__spaceItem_nr3as"
                >
                  <span
                    class="typography__primary-medium_c16y5 typography__bold_c16y5"
                  >
                     info@alfabankstore.ru
                  </span>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
              style="margin-bottom: 16px;"
            >
              <span
                class="typography__primary-medium_c16y5 typography__bold_c16y5"
              >
                г. Москва, пр-т Андропова, 18 корп. 3
              </span>
            </div>
            <div
              class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
              style="margin-bottom: 16px;"
            >
              <div
                class="space__spaceContainer_nr3as space__vertical_nr3as space__start_nr3as"
              >
                <div
                  class="space__spaceItem_nr3as"
                  style="margin-bottom: 0px;"
                >
                  <span
                    class="typography__primary-large_c16y5 typography__bold_c16y5"
                  >
                    пн-чт:
                  </span>
                </div>
                <div
                  class="space__spaceItem_nr3as"
                  style="margin-bottom: 0px;"
                >
                  <span
                    class="typography__primary-medium_c16y5 typography__bold_c16y5"
                  >
                    10:00—19:00
                  </span>
                </div>
                <div
                  class="space__spaceItem_nr3as"
                  style="margin-bottom: 0px;"
                >
                  <span
                    class="typography__primary-large_c16y5 typography__bold_c16y5"
                  >
                    пт:
                  </span>
                </div>
                <div
                  class="space__spaceItem_nr3as"
                >
                  <span
                    class="typography__primary-medium_c16y5 typography__bold_c16y5"
                  >
                     10:00—17:30
                  </span>
                </div>
              </div>
            </div>
            <div
              class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
            >
              <span
                class="typography__primary-medium_c16y5 typography__bold_c16y5"
              >
                 Принимаем к оплате карты Visa, Mastercard, МИР.
              </span>
            </div>
          </div>
        </div>
        <div
          class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
        >
          <div
            class="yandex_map"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
